awx/ui_next/src/routeConfig.jsx

```diff
@@ -155,6 +158,77 @@ function getRouteConfig() {
       ],
     },
   ];
+
+  const deleteRoute = name => {
+    routeConfig.forEach(group => {
+      group.routes = group.routes.filter(({ path }) => !path.includes(name));
+    });
+    routeConfig = routeConfig.filter(groups => groups.routes.length);
+  };
+
+  const deleteRouteGroup = name => {
+    routeConfig = routeConfig.filter(({ groupId }) => !groupId.includes(name));
+  };
+
+  if (!(me.is_superuser || me.is_system_auditor)) {
+    deleteRouteGroup('settings');
+
+    if (me.isOrgAdmin) {
+      ['credential_types', 'management_jobs'].forEach(path => {
+        deleteRoute(path);
+      });
+    }
+
+    if (!me.isOrgAdmin && !me.isExecEnvAdmin && me.isNotificationAdmin) {
+      [
+        'credential_types',
+        'management_jobs',
+        'instance_groups',
+        'applications',
+        'execution_environments',
+      ].forEach(path => {
+        deleteRoute(path);
+      });
+    }
+
+    if (!me.isOrgAdmin && me.isExecEnvAdmin && !me.isNotificationAdmin) {
+      [
+        'credential_types',
+        'notification_templates',
+        'management_jobs',
+        'instance_groups',
+        'applications',
+      ].forEach(path => {
+        deleteRoute(path);
+      });
+    }
+
+    if (!me.isOrgAdmin && me.isExecEnvAdmin && me.isNotificationAdmin) {
+      [
+        'credential_types',
+        'management_jobs',
+        'instance_groups',
+        'applications',
+      ].forEach(path => {
+        deleteRoute(path);
+      });
+    }
+
+    if (!me.isOrgAdmin && !me.isExecEnvAdmin && !me.isNotificationAdmin) {
+      [
+        'credential_types',
+        'notification_templates',
+        'management_jobs',
+        'instance_groups',
+        'applications',
+        'execution_environments',
+      ].forEach(path => {
+        deleteRoute(path);
+      });
+    }
+  }
```

Comment:
I think this can be simplified and if possible we should do so. The permissions are layered - As I see it, we need to loop over strings for each combination. The following should be equivalent:

```suggestion
  if (me?.is_superuser) return routeConfig;
  if (me?.is_system_auditor) return routeConfig;
  deleteRouteGroup('settings');
  deleteRoute('management_jobs');
  deleteRoute('credential_types');
  if (me?.isOrgAdmin) return routeConfig;
  deleteRoute('applications');
  deleteRoute('instance_groups');
  if (!me?.isNotificationAdmin) deleteRoute('notification_templates');
  if (!me?.isExecEnvAdmin) deleteRoute('execution_environments');
```

![Screenshot from 2021-06-02 11-26-54](https://user-images.githubusercontent.com/9753817/120508134-7840fb00-c395-11eb-9620-d8a1c29af8bc.png)

Reply:
@jakemcdermott updated. Great suggestion.